torch/_inductor/runtime/triton_bundler.py

```diff
@@ -0,0 +1,127 @@
+import dataclasses
+import logging
+import os
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import torch
+from torch._dynamo.utils import counters
+
+from .runtime_utils import cache_dir
+
+
+log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(frozen=True)
+class TritonBundleEntry:
+    kernel_hash: str
+    device: int
+    directory: str
+
+
+@dataclasses.dataclass
+class TritonKernelArtifact:
+    filename: str
+    payload: bytes = dataclasses.field(repr=False)  # Do not display binary
+
+
+@dataclasses.dataclass
+class TritonKernelArtifacts:
+    kernel_hash: str
+    device: int
+    artifacts: List[TritonKernelArtifact]
+
+
+class TritonBundler:
+    # It is safe for multiple threads to insert entries as insert operation
+    # of dict is atomic
+    _entries: Optional[Dict[TritonBundleEntry, None]] = None
```

Comment:
Nit - is this normal pytorch style? or should we just add a lock to make this thread safety explicit?

Reply:
The operation is atomic, why would we need a lock?

Reply:
Documentation. (also do you think the overhead will be significant here? If you do we can drop it, but assuming what 100k triton compiles, and 100ns for a mutex acquisition, you get 10ms in overhead).